arm/Microsoft.Network/virtualWans/deploy.bicep

```diff
@@ -0,0 +1,196 @@
+  'Resource Policy Contributor': subscriptionResourceId('Microsoft.Authorization/roleDefinitions','36243c78-bf99-498c-9df9-86d9f8d28608')
+  'User Access Administrator': subscriptionResourceId('Microsoft.Authorization/roleDefinitions','18d7d88d-d35e-4fb5-a5c3-7773c20a72d9')
+}
+
+module pid_cuaId './.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
+  name: 'pid-${cuaId}'
+  params: {}
+}
+
+resource virtualWan 'Microsoft.Network/virtualWans@2021-05-01' = {
+  name: wanName
+  location: location
+  tags: tags
+  properties: {
+    type: wanSku
+  }
+}
+
+resource vwan_lock 'Microsoft.Authorization/locks@2016-09-01' = if (lockForDeletion) {
+  name: '${wanName}-vwanDoNotDelete'
+  properties: {
+    level: 'CanNotDelete'
+  }
+  scope: virtualWan
+}
+
+resource virtualHub 'Microsoft.Network/virtualHubs@2021-05-01' = {
+  name: hubName
+  location: location
+  properties: {
+    addressPrefix: addressPrefix
+    virtualWan: {
+      id: virtualWan.id
+    }
+  }
+}
+
+resource virtualHub_lock 'Microsoft.Authorization/locks@2016-09-01' = if (lockForDeletion) {
+  name: '${hubName}-vHubDoNotDelete'
+  properties: {
+    level: 'CanNotDelete'
+  }
+  scope: virtualHub
+}
+
+resource vpnSite 'Microsoft.Network/vpnSites@2021-05-01' = {
+  name: vpnSiteName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: vpnsiteAddressspaceList
+    }
+    bgpProperties: {
+      asn: vpnsiteBgpAsn
+      bgpPeeringAddress: vpnsiteBgpPeeringAddress
+      peerWeight: 0
+    }
+    deviceProperties: {
+      linkSpeedInMbps: 0
+    }
+    ipAddress: vpnsitePublicIPAddress
+    virtualWan: {
+      id: virtualWan.id
+    }
+  }
+}
+
+resource vpnSite_lock 'Microsoft.Authorization/locks@2016-09-01' = if (lockForDeletion) {
+  name: '${vpnSiteName}-vpnSiteDoNotDelete'
+  properties: {
+    level: 'CanNotDelete'
+  }
+  scope: vpnSite
+}
+
+resource vpnGateway 'Microsoft.Network/vpnGateways@2021-05-01' = {
+  name: vpnGatewayName
+  location: location
+  properties: {
+    connections: [
+      {
+        name: connectionName
+        properties: {
+          connectionBandwidth: 10
+          enableBgp: any(enableBgp)
+          remoteVpnSite: {
+            id: vpnSite.id
+          }
+        }
+      }
+    ]
+    virtualHub: {
+      id: virtualHub.id
+    }
+    bgpSettings: {
+      asn: 65515
+    }
+  }
+}
+
+resource vpnGateway_lock 'Microsoft.Authorization/locks@2016-09-01' = if (lockForDeletion) {
+  name: '${vpnGatewayName}-vpnGatewayDoNotDelete'
+  properties: {
+    level: 'CanNotDelete'
+  }
+  scope: vpnGateway
+}
+
+module rbac_name './.bicep/nested_rbac.bicep' = [for (item, i) in roleAssignments: {
+  name: 'rbac-${deployment().name}${i}'
+  params: {
+    roleAssignment: item
+    builtInRoleNames: builtInRoleNames
+    wanName: wanName
+  }
+  dependsOn: [
+    virtualWan
+  ]
+}]
+
+output wanName string = wanName
+output wanNameResourceId string = virtualWan.id
```

Comment:
Should this not be virtualWanName, virtualWanResourceId and vWanResourceGroupName?
Also, should not virtualWanName string = virtualWan.name?

Reply:
Agreed. I just left it as it is.
Regarding the reference of the name either seems fine. Updated it now